packages/core/tests/unit/components/navi-visualizations/line-chart-test.ts

```diff
@@ -0,0 +1,711 @@
+          uniqueIdentifier: 183206656,
+          totalPageViews: 4088487125
+        },
+        {
+          'network.dateTime(grain=day)': '2016-06-01 00:00:00.000',
+          uniqueIdentifier: 183380921,
+          totalPageViews: 4024700302
+        },
+        {
+          'network.dateTime(grain=day)': '2016-06-02 00:00:00.000',
+          uniqueIdentifier: 180559793,
+          totalPageViews: 3950276031
+        },
+        {
+          'network.dateTime(grain=day)': '2016-06-03 00:00:00.000',
+          uniqueIdentifier: 172724594,
+          totalPageViews: 3697156058
+        }
+      ],
+      meta: {}
+    };
+    const request = buildTestRequest(
+      [{ field: 'uniqueIdentifier' }, { field: 'totalPageViews' }],
+      [],
+      { start: '2016-05-30 00:00:00.000', end: '2016-06-04 00:00:00.000' },
+      'day'
+    );
+
+    const model = { request, response };
+    const options = {
+      axis: {
+        y: {
+          series: {
+            type: 'metric',
+            config: {}
+          }
+        }
+      }
+    };
+    let component = createGlimmerComponent('component:navi-visualizations/line-chart', {
+      model: A([model]),
+      options
+    }) as LineChart;
+
+    component.dataConfig; // since we're unit testing the component, this line is needed
+
+    // get tooltip and see if it has the right rowData
+    let tooltipComp = component.tooltipComponent;
+    let tooltip = tooltipComp.create({
+      tooltipData: [
+        {
+          name: 'Unique Identifiers',
+          id: 'Unique Identifiers',
+          index: 5,
+          seriesIndex: 0,
+          value: 172933788,
+          x: '2016-06-03 00:00:00.000'
+        }
+      ],
+      x: '2016-06-03 00:00:00.000',
+      seriesConfig: component.seriesConfig,
+      request
+    });
+    run(() => {
+      let element = document.createElement('div');
+      tooltip.appendTo(element);
+    });
+    assert.ok(
+      tooltip.rowData.firstObject.hasOwnProperty('uniqueIdentifier'),
+      'Initial tooltip render has the right rowData'
+    );
+
+    //new data
+    let response2 = {
+      rows: [
+        {
+          'network.dateTime(grain=day)': '2016-05-30 00:00:00.000',
+          navClicks: 172933788
+        },
+        {
+          'network.dateTime(grain=day)': '2016-05-31 00:00:00.000',
+          navClicks: 4088487125
+        },
+        {
+          'network.dateTime(grain=day)': '2016-06-01 00:00:00.000',
+          navClicks: 183380921
+        },
+        {
+          'network.dateTime(grain=day)': '2016-06-02 00:00:00.000',
+          navClicks: 3950276031
+        },
+        {
+          'network.dateTime(grain=day)': '2016-06-03 00:00:00.000',
+          navClicks: 172724594
+        }
+      ],
+      meta: {}
+    };
+
+    set(component.args, 'model', A([{ request, response: response2 }]));
+
+    component.dataConfig; // since we're unit testing the component, this line is needed
```

Comment:
Maybe we should come back to this. It doesn't seem like a great pattern that we have a getter that causes side effects that other properties depend on